openedx/core/lib/block_structure/transformer.py

```diff
@@ -125,4 +126,55 @@ def transform(self, usage_info, block_structure):
                 block structure, with already collected data for the
                 transformer, that is to be transformed in place.
         """
-        pass
+        raise NotImplementedError
+
+
+class OptimizedTransformer(BlockStructureTransformer):
```

Comment:
Not sure that would work - since different transformers require different traversal algorithms.

I think a `Mixin` seems appropriate here.